app/cdap/components/DirectiveInput/index.tsx

```diff
@@ -0,0 +1,179 @@
+  bottom: 50px;
+  width: 100%;
+`;
+
+const CloseIconButton = styled(IconButton)`
+  color: #ffffff;
+`;
+
+export default function ({
+  columnNamesList,
+  onDirectiveInputHandler,
+  onClose,
+  openDirectivePanel,
+}: IDirectiveInputProps) {
+  const [inputBoxValue, setInputBoxValue] = useState<string>('');
+  const [columnSelected, setColumnSelected] = useState<boolean>(false);
+  const [selectedDirective, setSelectedDirective] = useState<boolean>(false);
+  const [directiveUsage, setDirectiveUsage] = useState<IDirectiveUsage[]>([]);
+  const [directivesList, setDirectivesList] = useState<IDirectivesList[]>([]);
+  const directiveRef = useRef();
+
+  const handleDirectiveChange = (value: string) => {
+    !value && setSelectedDirective(false);
+    setInputBoxValue(value);
+  };
+
+  useEffect(() => {
+    setDirectiveUsage(formatDirectiveUsageData(inputBoxValue, directivesList));
+  }, [inputBoxValue]);
+
+  const handleKeyDownEvent = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    {
+      const directiveSyntax =
+        directiveUsage.length > 0 ? directiveUsage[0]?.item?.usage.split(' ') : [];
+      const inputSplit = inputBoxValue.replace(/^\s+/g, '').split(' ');
+      console.log('directiveSyntax, inputSplit', directiveSyntax, inputSplit)
+      if (event.key === 'Enter' && handlePasteDirective(inputBoxValue, directivesList)) {
+        onDirectiveInputHandler(inputBoxValue);
+      }
+      else if (multipleColumnDirective.includes(inputSplit[0]) && event.key === 'Enter' &&
+        columnSelected &&
+        selectedDirective && (directiveSyntax.length === inputSplit.length || inputSplit.length > directiveSyntax.length || inputSplit.length < directiveSyntax.length)) {
+        onDirectiveInputHandler(inputBoxValue);
+      }
+      else if (
+        event.key === 'Enter' &&
+        columnSelected &&
+        selectedDirective &&
+        (directiveSyntax.length === inputSplit.length || inputSplit.length > directiveSyntax.length)
+      ) {
+        onDirectiveInputHandler(inputBoxValue);
+      }
+    }
+  };
+
+  return (
+    <DirectiveBox data-testid="directive-input-main-container">
+      {openDirectivePanel && (
+        <SimpleBox data-testid="directive-input-parent">
+          <InputPanel
+            inputBoxValue={inputBoxValue}
+            onSearchItemClick={(value) => handleDirectiveChange(value)}
+            setDirectivesList={setDirectivesList}
+            getDirectiveSyntax={(activeResults: IDirectiveUsage[], value) => {
+              setSelectedDirective(value);
+              setDirectiveUsage(activeResults);
+            }}
+            onColumnSelection={() => {
+              setColumnSelected(true);
+            }}
+            selectedDirective={selectedDirective}
+            columnNamesList={columnNamesList}
+          />
+          <DirectiveUsageWrapper>
+            {directiveUsage.length === 1 &&
+              directiveUsage.map((eachDirective: IDirectiveUsage) => (
```

Comment:
again here this doesn't really make sense contextually - why isn't this something like directiveUsages or directive useageList and then directiveUsage inside the map?

Reply:
Fixed